pkg/service/keystore/service.go

```diff
@@ -50,20 +49,17 @@ func (s Service) Config() config.KeyStoreServiceConfig {
 }
 
 func NewKeyStoreService(config config.KeyStoreServiceConfig, s storage.ServiceStorage) (*Service, error) {
-	if err := EnsureServiceKeyExists(config, s); err != nil {
-		return nil, sdkutil.LoggingErrorMsg(err, "initializing keystore")
+	encrypter, decrypter, err := NewServiceEncryption(s, config.EncryptionConfig, ServiceKeyEncryptionKey)
```

Comment:
I would prefer, that instead of a `noopEncrypter` and `noopDecrypter` we check config here, and if the flag is off we pass in nil values below, this avoids the overhead of a no-op call for each read/write